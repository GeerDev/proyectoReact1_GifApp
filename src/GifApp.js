import React, {useState} from 'react'; 

import {AddCategory} from './components/AddCategory'
import { GifGrid } from './components/GifGrid';

export const GifApp = ({ defaultCategories = [] }) => {

	const [categorias, setcategorias] = useState( defaultCategories )
	
	return (
	   <> 
		<h2>Gifs App</h2>
		<AddCategory propiedad = { setcategorias } />
		<hr/>

		<ol>
		{

			categorias.map (category => (
				<GifGrid 
				key={ category }
				category={ category }
				/>
				))

		}
		</ol>
	   </>
		);

}
