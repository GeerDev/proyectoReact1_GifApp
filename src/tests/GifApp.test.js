import React from 'react';
import { shallow } from 'enzyme';
import { GifApp } from '../GifApp';

describe('Pruebas en <GifApp />', () => {
    
    test('debe mostrarse correctamente', () => {

        const wrapper = shallow(<GifApp /> );
        expect( wrapper ).toMatchSnapshot();
        
    });

    test('debe de mostrar una lista de categorias', () => {

        const categories = ['One Punch', 'Dragon Ball'];
        const wrapper = shallow(<GifApp defaultCategories={ categories } /> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );

    })
    
})