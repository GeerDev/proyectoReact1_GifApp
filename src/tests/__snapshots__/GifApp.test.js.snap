// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifApp /> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    Gifs App
  </h2>
  <AddCategory
    propiedad={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en <GifApp /> debe mostrarse correctamente 1`] = `
<Fragment>
  <h2>
    Gifs App
  </h2>
  <AddCategory
    propiedad={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
